codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -429,7 +429,11 @@ private void readHeadersFrame(final ChannelHandlerContext ctx, ByteBuf payload,
             final boolean exclusive = (word1 & 0x80000000L) != 0;
             final int streamDependency = (int) (word1 & 0x7FFFFFFFL);
             if (streamDependency == streamId) {
-                throw streamError(streamId, PROTOCOL_ERROR, "A stream cannot depend on itself.");
+                // RFC 7540 section 5.3.1 says this must be treated as a stream error of type PROTOCOL_ERROR
```

Comment:
Consider using RFC 9113

Reply:
RFC 9113 deprecated stream dependencies and unfortunately refers readers to RFC 7540 to find out the details.

Reply:
I've updated the comment to include this nuance.